instance/models/instance.py

```diff
@@ -207,52 +205,61 @@ def _sort_log_entries(server_logs, instance_logs):
         next_server_log_entry = partial(next, server_logs, None)
         next_instance_log_entry = partial(next, instance_logs, None)
 
-        log = []
         instance_log_entry = next_instance_log_entry()
         server_log_entry = next_server_log_entry()
 
         while instance_log_entry is not None and server_log_entry is not None:
             if server_log_entry.created < instance_log_entry.created:
-                log.append(server_log_entry)
+                yield server_log_entry
                 server_log_entry = next_server_log_entry()
             else:
-                log.append(instance_log_entry)
+                yield instance_log_entry
                 instance_log_entry = next_instance_log_entry()
 
         while instance_log_entry is not None:
-            log.append(instance_log_entry)
+            yield instance_log_entry
             instance_log_entry = next_instance_log_entry()
 
         while server_log_entry is not None:
-            log.append(server_log_entry)
+            yield server_log_entry
             server_log_entry = next_server_log_entry()
```

Comment:
For the time being, let's just leave the code the way it is.  While it's easy to write a helper function implements the `key` parameter around the old `heapq.merge()`, it's rather cumbersome (and slow) to write something that supports the `reversed` parameter.  For reversing, you really need to implement custom comparisons, and can't simply annotate the objects with their keys.

We can clean this up once we're using Python 3.5.